### DISCUSSION TITLE
change icon of all window contacts depending on state

### DISCUSSION THREAD

**Discussion by bhessen** [Mariusthvdb/custom-ui#140] [Q&A] [answered]
Hi, I tried to follow the examples and change the icon of all my window contacts (and also the icon color) depending on the state. I added custom-ui from HACS and added those lines to configuration.yaml:
`homeassistant:`
`  customize_glob:`
`    binary_sensor.fensterkontakt_*:`
`      templates:`
`        icon: >`
`          if (state === 'on') return mdi:window-open-variant;`
`          return mdi:window-closed-variant;`
`        icon_color: >`
`          if (state === 'on') return 'blue'; `
`          return 'grey';`

But all this is doing is this:
![image](https://github.com/Mariusthvdb/custom-ui/assets/8470814/9a20e03a-c846-4be9-8e41-1bbd90328f12)

The icon and the color of the entities itsself are not changed. What am I missing here?

**Comment by Mariusthvdb** [comment]
not sure. some items can not be customized like this, depending on their source. what integration made these entities?

can you confirm you see the resource being installed in the inspector Console panel ?
templates seem fine.

btw, You can also color those binary_sensor icons using the official [state-colors set in a theme (the preferred way)](https://www.home-assistant.io/integrations/frontend/#state-color)

is there no device_class on these entities? that would solve all of your needs.

set the` device_class: cover` or window on them, you can also do that via customize.

**Comment by bhessen** [comment, marked as answer]
Solved it: I was missing the quotation marks in the return statement for the icon:
`if (state === 'on') return 'mdi:window-open-variant';`

device_class was already set for these entities. But the window device class uses the mdi:window-open/closed icon. I want to have the mdi:window-open/closed-variant icon for all my window entities

**Reply by Mariusthvdb** [reply]
quotation marks ...


sorry, I should have spotted that. yes, always required inside templates..

**Comment by Mariusthvdb** [comment]
so this can be closed?
